commands/base_command.go

```diff
@@ -128,13 +128,8 @@ func GetOptionValue(args []string, optionName string) string {
 }
 
 // NewRestClient creates a new MTA deployer REST client
-func (c *BaseCommand) NewRestClient(host string) (restclient.RestClientOperations, error) {
-	restClient := c.clientFactory.NewRestClient(host, c.transport, c.jar, c.tokenFactory)
-	return restClient, nil
-}
-
-func (c *BaseCommand) NewManagementRestClient(host string) (restclient.RestClientOperations, error) {
-	return c.clientFactory.NewManagementRestClient(host, c.transport, c.jar, c.tokenFactory), nil
+func (c *BaseCommand) NewRestClient(host string) restclient.RestClientOperations {
+	return c.clientFactory.NewRestClient(host, c.transport, c.jar, c.tokenFactory)
```

Comment:
As I can see the practise is to return client and nil as second value which indicate that there is no errors. Take a look at NewMtaClient and NewMtaV2Client functions

Reply:
I don't think it's a "convention" per se, it's just that, in the `NewMtaClient` and `NewMtaV2Client` methods, there is an operation that returns an error (the `GetSpace` call), whereas here, an error can't be returned when creating the client.